app/controllers/api/authentications_controller.rb

```diff
@@ -1,4 +1,27 @@
 module Api
   class AuthenticationsController < BaseController
+    def create_resource(_type, _id, data)
+      attrs = validate_auth_attrs(data)
+      klass = ::Authentication.class_from_request_data(attrs)
+      # TODO: Temporary validation - remove
+      raise 'type not currently supported' unless klass.respond_to?(:create_in_provider_queue)
+      task_id = klass.create_in_provider_queue(attrs['manager_resource'], attrs.except('type', 'manager_resource'))
+      action_result(true, 'Creating Authentication', :task_id => task_id)
+    rescue => err
+      action_result(false, err.to_s)
+    end
+
+    private
+
+    def validate_auth_attrs(data)
+      raise 'must supply a manager resource' unless data['manager_resource']
+      attrs = data.dup
+      attrs['manager_resource'] = if data['manager_resource'].key?('href')
+                                    parse_href(data['manager_resource']['href']).last
+                                  elsif data['manager_resource'].key?('id')
+                                    data['manager_resource']['id']
+                                  end
```

Comment:
couple of questions here.

what is the expected value of manager_resource for create_in_provider_queue ? Is it an id ? if so an id to which class ?

parse_href returns collection, id.  If user sends in an href to a different  resource, say /api/users/10,  that means we're sending in a bogus 10.  In which case could  create_in_provider_queue supports an object for the first parameter ?  

what is the else case, user specified "manager_resource" : 10 ?